### PR TITLE
53_杨楫_第三次作业

### PR DESCRIPTION
**Pull request by yyssjj33**
@taxic

### CODE REVIEW

File: Lesson-3/assignment/yours.sol [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+=========================================================================================
+第三题（加分题）：自学C3 Linearization, 求以下 contract Z 的继承线                            |
+contract O                                                                              |
+contract A is O                                                                         |
+contract B is O                                                                         |
+contract C is O                                                                         |
+contract K1 is A, B                                                                     |
+contract K2 is A, C                                                                     |
+contract Z is K1, K2                                                                    |
+=========================================================================================
```
Comment (zchcai) [inline Lesson-3/assignment/yours.sol:151]: L(O) = [O]
L(A) = [A, O]
L(B) = [B, O]
L(C) = [C, O]
L(K2) = [K2] + merge(L(C), L(A), [C, A])
     = [K2] + merge([C, O], [A, O], [C, A])
     = [K2, C] + merge([O], [A, O], [A])
     = [K2, C, A, O]
L(K1) = [K1] + merge(L(B), L(A), [B, A])
     = [K1] + merge([B, O], [A, O], [B, A])
     = [K1, B] + merge([O], [A, O], [A])
     = [K1, B, A, O]
     
L(Z) = [Z] + merge(L(K2), L(K1), [K2, K1])
    = [Z] + merge([K2, C, A, O], [K1, B, A, O], [K2, K1])
    = [Z, K2] + merge([C, A, O], [K1, B, A, O], [K1])
    = [Z, K2, C] + merge([A, O], [K1, B, A, O], [K1])
    = [Z, K2, C, K1] + merge([A, O], [B, A, O])
    = [Z, K2, C, K1, B] + merge([A, O], [A, O])
    = [Z, K2, C, K1, B, A, O]

File: Lesson-3/assignment/yours.sol [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+https://github.com/yyssjj33/Team-F/raw/53-%E6%9D%A8%E6%A5%AB/Lesson-3/assignment/%E4%BD%9C%E4%B8%9A%E4%B8%89%E7%AC%AC%E4%B8%80%E9%A2%98%E6%88%AA%E5%9B%BE.zip
+
+
+
+=========================================================================================
+第二题：增加 changePaymentAddress 函数，更改员工的薪水支付地址，思考一下能否使用modifier整合某个功能|
+=========================================================================================
+可以整合partialPaid函数到modifier
+modifier partialPaid(employeeId) {
```
Comment (zchcai) [inline Lesson-3/assignment/yours.sol]: using partialPay as modifier is interesting. try to add employeeNonExist as well

**Review by zchcai — COMMENTED** [review]
minor fix needs

**Comment by yyssjj33** [comment]
@taxic made some review changes

File: Lesson-3/assignment/yours.sol [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        var employee = employees[employeeId];
+        
+        totalSalary -= employee.salary;
+        totalSalary += s * 1 ether;
+        employees[employeeId].salary = s * 1 ether;
+        employees[employeeId].lastPayday = now;
+    }
+    
+    function changePaymentAddress(address oldAddress, address newAddress) onlyOwner employeeExist(oldAddress) employeeNonExist(newAddress) partialPaid(oldAddress){
+        var employee = employees[oldAddress];
```
Comment (zchcai) [inline Lesson-3/assignment/yours.sol]: require new Address not zero

**Review by zchcai — CHANGES_REQUESTED** [review]

**Comment by yyssjj33** [comment]
@taxic 已修改！

**Review by zchcai — APPROVED** [review]

**Pull request merged**